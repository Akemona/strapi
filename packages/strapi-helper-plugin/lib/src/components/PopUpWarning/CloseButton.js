import styled from 'styled-components';

const CloseButton = styled.div`
  position: absolute;
  top: 18px;
  right: 30px;
  cursor: pointer;
  > svg {
    fill: #c3c5c8;
  }
`;

export default CloseButton;
