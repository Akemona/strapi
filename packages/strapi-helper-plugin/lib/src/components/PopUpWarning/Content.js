import styled from 'styled-components';
import { Text } from '@buffetjs/core';

const Content = styled(Text)`
  width: ${({ small }) => (small ? '200px' : '100%')};
  line-height: 1.8rem;
  margin: auto;
  min-height: 36px;
`;

Content.defaultProps = {
  lineHeight: '18px',
  color: 'lightOrange',
};

export default Content;
