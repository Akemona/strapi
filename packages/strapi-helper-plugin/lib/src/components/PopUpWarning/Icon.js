import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import IcoDanger from '../../assets/icons/icon_danger.svg';
import IcoNotFound from '../../assets/icons/icon_flag_not_found.svg';
import IcoInfo from '../../assets/icons/icon_info.svg';
import IcoSuccess from '../../assets/icons/icon_success.svg';
import IcoWarning from '../../assets/icons/icon_warning.svg';

const icons = {
  danger: IcoDanger,
  info: IcoInfo,
  notFound: IcoNotFound,
  success: IcoSuccess,
  warning: IcoWarning,
};

const Img = styled.img`
  width: 2.5rem;
  margin-bottom: 2.2rem;
`;

const Icon = ({ type }) => {
  return <Img src={icons[type]} alt="icon" />;
};

Icon.propTypes = {
  type: PropTypes.string.isRequired,
};

export default memo(Icon);
