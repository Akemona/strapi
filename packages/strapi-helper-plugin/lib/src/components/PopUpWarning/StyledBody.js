import styled from 'styled-components';
import { ModalBody } from 'reactstrap';

const StyledBody = styled(ModalBody)`
  padding: 0;
  text-align: center;
  > div {
    padding: 2.1rem 3rem 1.5rem 3rem !important;
    text-align: center;
  }
`;

export default StyledBody;
