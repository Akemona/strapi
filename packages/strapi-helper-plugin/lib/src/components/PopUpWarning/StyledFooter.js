import styled from 'styled-components';

const StyledFooter = styled.div`
  display: flex;
  width: 100%;
  margin-top: 19px;
  justify-content: space-between;
  background-color: #eff3f6;
  padding: 15px 30px 17px 30px;
  > button {
    padding-left: 30px;
    padding-right: 30px;
  }
`;

export default StyledFooter;
