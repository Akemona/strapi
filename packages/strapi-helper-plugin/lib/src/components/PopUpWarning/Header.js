import React from 'react';
import { FormattedMessage } from 'react-intl';
import Close from '../../svgs/Close';
import CloseButton from './CloseButton';
import StyledHeader from './StyledHeader';

const Header = ({ onClick, title, children }) => {
  return (
    <>
      <CloseButton onClick={onClick}>
        <Close fill="#c3c5c8" />
      </CloseButton>
      <StyledHeader toggle={onClick}>
        {children || <FormattedMessage id={title || 'components.popUpWarning.title'} />}
      </StyledHeader>
    </>
  );
};

export default Header;
