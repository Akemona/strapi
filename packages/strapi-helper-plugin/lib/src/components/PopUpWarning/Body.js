import React, { memo } from 'react';
import PropTypes from 'prop-types';
import StyledBody from './StyledBody';

const Body = ({ children }) => (
  <StyledBody>
    <div>{children}</div>
  </StyledBody>
);

Body.propTypes = {
  children: PropTypes.node.isRequired,
};

export default memo(Body);
