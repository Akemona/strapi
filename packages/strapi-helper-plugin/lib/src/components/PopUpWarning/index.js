/**
 *
 * PopUpWarning
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { map } from 'lodash';
import { Button, Padded, Text } from '@buffetjs/core';
import { FormattedMessage } from 'react-intl';
import Body from './Body';
import ContentText from './Content';
import Header from './Header';
import Icon from './Icon';
import StyledModal from './StyledModal';
import StyledFooter from './StyledFooter';

function PopUpWarning({
  content,
  isOpen,
  isConfirmButtonLoading,
  onConfirm,
  onlyConfirmButton,
  popUpWarningType,
  toggleModal,
  children,
  ...rest
}) {
  const handleToggle = e => {
    // Prevent user interactions while requests are being submitted
    if (isConfirmButtonLoading) {
      return;
    }

    toggleModal(e);
  };

  const buttons = [
    {
      color: 'cancel',
      onClick: handleToggle,
      message: content.cancel || 'components.popUpWarning.button.cancel',
    },
    {
      color: 'delete',
      onClick: onConfirm,
      isLoading: isConfirmButtonLoading,
      message: content.confirm || 'components.popUpWarning.button.confirm',
    },
  ];

  const singleButton = [
    {
      color: 'delete',
      onClick: onConfirm,
      message: content.confirm || 'components.popUpWarning.button.confirm',
      style: { width: '100%' },
    },
  ];
  const footerButtons = onlyConfirmButton ? singleButton : buttons;

  return (
    <StyledModal isOpen={isOpen} toggle={handleToggle} {...rest}>
      <Header onClick={handleToggle} title={content.title} />
      <Body>
        <Icon type={popUpWarningType} />
        <ContentText small={content.secondMessage}>
          <FormattedMessage
            id={content.message || 'components.popUpWarning.message'}
            values={content.messageValues}
          />
        </ContentText>
        {content.secondMessage && (
          <Padded top size="smd">
            <Text color="lightOrange">
              <FormattedMessage id={content.secondMessage} />
            </Text>
          </Padded>
        )}
        {children}
      </Body>
      <StyledFooter>
        {map(footerButtons, button => {
          const { message, onClick, ...rest } = button;
          return (
            <Button key={message} onClick={onClick} {...rest}>
              <FormattedMessage id={message} />
            </Button>
          );
        })}
      </StyledFooter>
    </StyledModal>
  );
}

PopUpWarning.propTypes = {
  content: PropTypes.shape({
    cancel: PropTypes.string,
    confirm: PropTypes.string,
    message: PropTypes.string,
    secondMessage: PropTypes.string,
    title: PropTypes.string,
  }),
  isConfirmButtonLoading: PropTypes.bool,
  isOpen: PropTypes.bool.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onlyConfirmButton: PropTypes.bool,
  popUpWarningType: PropTypes.string,
  toggleModal: PropTypes.func.isRequired,
};

PopUpWarning.defaultProps = {
  content: {
    cancel: 'components.popUpWarning.button.cancel',
    confirm: 'components.popUpWarning.button.confirm',
    message: 'components.popUpWarning.message',
    messageValues: {},
    secondMessage: null,
    title: 'components.popUpWarning.title',
  },
  isConfirmButtonLoading: false,
  onlyConfirmButton: false,
  popUpWarningType: 'danger',
};

export default PopUpWarning;
