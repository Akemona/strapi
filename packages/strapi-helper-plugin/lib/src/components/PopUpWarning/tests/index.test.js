import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { fireEvent } from '@testing-library/react';

import renderWithIntl from '../../../testUtils/renderWithIntl';
import PopUpWarning from '../index';

const translationMessages = {
  'components.popUpWarning.button.cancel': 'Cancel',
  'components.popUpWarning.button.confirm': 'Confirm',
  'components.popUpWarning.message': 'Are you sure you want to delete this?',
  'components.popUpWarning.title': 'Please confirm',
};

describe('<PopUpWarning />', () => {
  const props = {
    content: {
      cancel: 'components.popUpWarning.button.cancel',
      confirm: 'components.popUpWarning.button.confirm',
      message: 'components.popUpWarning.message',
      title: 'components.popUpWarning.title',
    },
    isOpen: true,
    onConfirm: jest.fn(),
    onlyConfirmButton: false,
    popUpWarningType: 'warning',
    toggleModal: jest.fn(),
  };

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  describe('should render properly', () => {
    it('should not crash', () => {
      shallow(<PopUpWarning {...props} />);
    });

    it('should match snapshot if onlyConfirmButton is false', () => {
      const { asFragment } = renderWithIntl(<PopUpWarning {...props} />, translationMessages);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should match snapshot if onlyConfirmButton is true', () => {
      const { asFragment } = renderWithIntl(
        <PopUpWarning {...props} onlyConfirmButton={true} />,
        translationMessages
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('should call methods properly', () => {
    it('should call onConfirm props on confirm button click', () => {
      const { getByText } = renderWithIntl(<PopUpWarning {...props} />, translationMessages);

      const node = getByText('Confirm');
      fireEvent.click(node);

      expect(props.onConfirm).toHaveBeenCalled();
    });

    it('should call toggleModal props on cancel button click', () => {
      const { getByText } = renderWithIntl(<PopUpWarning {...props} />, translationMessages);

      const node = getByText('Cancel');
      fireEvent.click(node);

      expect(props.toggleModal).toHaveBeenCalled();
    });
  });
});
