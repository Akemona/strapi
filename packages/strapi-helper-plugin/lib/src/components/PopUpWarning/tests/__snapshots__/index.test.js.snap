// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopUpWarning /> should render properly should match snapshot if onlyConfirmButton is false 1`] = `
<DocumentFragment>
  .c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 15px;
  font-weight: 600;
  font-size: 1.3rem;
  border-radius: 2px;
  cursor: pointer;
  outline: 0;
  background-color: #ffffff;
  background-color: transparent;
  border: 1px solid #9ea7b8;
  color: #9ea7b8;
}

.c8:hover,
.c8:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c8:focus {
  outline: 0;
}

.c8 > span svg {
  font-size: 10px;
}

.c8 > svg {
  height: 10px;
  width: auto;
  margin-right: 10px;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 15px;
  font-weight: 600;
  font-size: 1.3rem;
  border-radius: 2px;
  cursor: pointer;
  outline: 0;
  background-color: #ffffff;
  background-color: #F64D0A;
  border: 1px solid #F64D0A;
  color: #ffffff;
}

.c9:hover,
.c9:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c9:focus {
  outline: 0;
}

.c9 > span svg {
  font-size: 10px;
}

.c9 > svg {
  height: 10px;
  width: auto;
  margin-right: 10px;
}

.c5 {
  margin: 0;
  line-height: 18px;
  color: #f64d0a;
  font-size: 13px;
  font-weight: 400;
  text-transform: none;
}

.c3 {
  padding: 0;
  text-align: center;
}

.c3 > div {
  padding: 2.1rem 3rem 1.5rem 3rem !important;
  text-align: center;
}

.c6 {
  width: 100%;
  line-height: 1.8rem;
  margin: auto;
  min-height: 36px;
}

.c1 {
  position: absolute;
  top: 18px;
  right: 30px;
  cursor: pointer;
}

.c1 > svg {
  fill: #c3c5c8;
}

.c2 {
  padding-left: 30px;
  padding-right: 30px;
  padding-top: 17px;
  padding-bottom: 22px !important;
  border: 0;
  background-color: #fafafa;
}

.c2 > h5 {
  width: 100%;
  text-align: center;
  font-family: Lato;
  font-weight: bold !important;
  font-size: 1.8rem !important;
  line-height: normal;
}

.c2 > button {
  display: none;
}

.c4 {
  width: 2.5rem;
  margin-bottom: 2.2rem;
}

.c0 {
  width: 41.6rem !important;
  margin: 14.4rem auto !important;
  -webkit-font-smoothing: antialiased !important;
}

.c0 > div {
  width: 41.6rem;
  padding: 0 !important;
  border: none;
  border-radius: 2px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-top: 19px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #eff3f6;
  padding: 15px 30px 17px 30px;
}

.c7 > button {
  padding-left: 30px;
  padding-right: 30px;
}

<div
    class=""
  >
    <div
      class="modal fade"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog c0"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="c1"
          >
            <svg
              fill="#c3c5c8"
              height="10px"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.0719417,0.127226812 C10.1612888,0.127226812 10.2403266,0.161591074 10.3090551,0.230319596 L10.3090551,0.230319596 L10.8245191,0.745783513 C10.8932476,0.814512036 10.9276118,0.893549837 10.9276118,0.982896916 C10.9276118,1.07224399 10.8932476,1.1512818 10.8245191,1.22001032 L10.8245191,1.22001032 L6.77297267,5.27155671 L10.8245191,9.3231031 C10.8932476,9.39183162 10.9276118,9.47086942 10.9276118,9.5602165 C10.9276118,9.64956358 10.8932476,9.72860138 10.8245191,9.79732991 L10.8245191,9.79732991 L10.3090551,10.3127938 C10.2403266,10.3815223 10.1612888,10.4158866 10.0719417,10.4158866 C9.98259466,10.4158866 9.90355686,10.3815223 9.83482834,10.3127938 L9.83482834,10.3127938 L5.92809485,6.40509433 C4.98802554,7.34516364 3.68545904,8.64773014 2.02039535,10.3127938 C1.95166683,10.3815223 1.87262903,10.4158866 1.78328195,10.4158866 C1.69393487,10.4158866 1.61489707,10.3815223 1.54616855,10.3127938 L1.03070463,9.79732991 C0.961976106,9.72860138 0.927611845,9.64956358 0.927611845,9.5602165 C0.927611845,9.47086942 0.961976106,9.39183162 1.03070463,9.3231031 L5.08225102,5.27155671 L1.03070463,1.22001032 C0.961976106,1.1512818 0.927611845,1.07224399 0.927611845,0.982896916 C0.927611845,0.893549837 0.961976106,0.814512036 1.03070463,0.745783513 L1.54616855,0.230319596 C1.61489707,0.161591074 1.69393487,0.127226812 1.78328195,0.127226812 C1.87262903,0.127226812 1.95166683,0.161591074 2.02039535,0.230319596 L5.92761184,4.13822681 L9.83482834,0.230319596 C9.88637473,0.178773204 9.94372009,0.146556709 10.0068644,0.133670111 Z"
                fill-rule="nonzero"
              />
            </svg>
          </div>
          <div
            class="c2 modal-header"
          >
            <h5
              class="modal-title"
            >
              <span>
                Please confirm
              </span>
            </h5>
            <button
              aria-label="Close"
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            class="c3 modal-body"
          >
            <div>
              <img
                alt="icon"
                class="c4"
                src="IMAGE_MOCK"
              />
              <p
                class="c5 c6"
                color="lightOrange"
                font-size="md"
                font-weight="regular"
              >
                <span>
                  Are you sure you want to delete this?
                </span>
              </p>
            </div>
          </div>
          <div
            class="c7"
          >
            <button
              class="c8"
              color="cancel"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="c9"
              color="delete"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade"
    />
  </div>
</DocumentFragment>
`;

exports[`<PopUpWarning /> should render properly should match snapshot if onlyConfirmButton is true 1`] = `
<DocumentFragment>
  .c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 15px;
  font-weight: 600;
  font-size: 1.3rem;
  border-radius: 2px;
  cursor: pointer;
  outline: 0;
  background-color: #ffffff;
  background-color: #F64D0A;
  border: 1px solid #F64D0A;
  color: #ffffff;
}

.c8:hover,
.c8:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c8:focus {
  outline: 0;
}

.c8 > span svg {
  font-size: 10px;
}

.c8 > svg {
  height: 10px;
  width: auto;
  margin-right: 10px;
}

.c5 {
  margin: 0;
  line-height: 18px;
  color: #f64d0a;
  font-size: 13px;
  font-weight: 400;
  text-transform: none;
}

.c3 {
  padding: 0;
  text-align: center;
}

.c3 > div {
  padding: 2.1rem 3rem 1.5rem 3rem !important;
  text-align: center;
}

.c6 {
  width: 100%;
  line-height: 1.8rem;
  margin: auto;
  min-height: 36px;
}

.c1 {
  position: absolute;
  top: 18px;
  right: 30px;
  cursor: pointer;
}

.c1 > svg {
  fill: #c3c5c8;
}

.c2 {
  padding-left: 30px;
  padding-right: 30px;
  padding-top: 17px;
  padding-bottom: 22px !important;
  border: 0;
  background-color: #fafafa;
}

.c2 > h5 {
  width: 100%;
  text-align: center;
  font-family: Lato;
  font-weight: bold !important;
  font-size: 1.8rem !important;
  line-height: normal;
}

.c2 > button {
  display: none;
}

.c4 {
  width: 2.5rem;
  margin-bottom: 2.2rem;
}

.c0 {
  width: 41.6rem !important;
  margin: 14.4rem auto !important;
  -webkit-font-smoothing: antialiased !important;
}

.c0 > div {
  width: 41.6rem;
  padding: 0 !important;
  border: none;
  border-radius: 2px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-top: 19px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #eff3f6;
  padding: 15px 30px 17px 30px;
}

.c7 > button {
  padding-left: 30px;
  padding-right: 30px;
}

<div
    class=""
  >
    <div
      class="modal fade"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog c0"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="c1"
          >
            <svg
              fill="#c3c5c8"
              height="10px"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.0719417,0.127226812 C10.1612888,0.127226812 10.2403266,0.161591074 10.3090551,0.230319596 L10.3090551,0.230319596 L10.8245191,0.745783513 C10.8932476,0.814512036 10.9276118,0.893549837 10.9276118,0.982896916 C10.9276118,1.07224399 10.8932476,1.1512818 10.8245191,1.22001032 L10.8245191,1.22001032 L6.77297267,5.27155671 L10.8245191,9.3231031 C10.8932476,9.39183162 10.9276118,9.47086942 10.9276118,9.5602165 C10.9276118,9.64956358 10.8932476,9.72860138 10.8245191,9.79732991 L10.8245191,9.79732991 L10.3090551,10.3127938 C10.2403266,10.3815223 10.1612888,10.4158866 10.0719417,10.4158866 C9.98259466,10.4158866 9.90355686,10.3815223 9.83482834,10.3127938 L9.83482834,10.3127938 L5.92809485,6.40509433 C4.98802554,7.34516364 3.68545904,8.64773014 2.02039535,10.3127938 C1.95166683,10.3815223 1.87262903,10.4158866 1.78328195,10.4158866 C1.69393487,10.4158866 1.61489707,10.3815223 1.54616855,10.3127938 L1.03070463,9.79732991 C0.961976106,9.72860138 0.927611845,9.64956358 0.927611845,9.5602165 C0.927611845,9.47086942 0.961976106,9.39183162 1.03070463,9.3231031 L5.08225102,5.27155671 L1.03070463,1.22001032 C0.961976106,1.1512818 0.927611845,1.07224399 0.927611845,0.982896916 C0.927611845,0.893549837 0.961976106,0.814512036 1.03070463,0.745783513 L1.54616855,0.230319596 C1.61489707,0.161591074 1.69393487,0.127226812 1.78328195,0.127226812 C1.87262903,0.127226812 1.95166683,0.161591074 2.02039535,0.230319596 L5.92761184,4.13822681 L9.83482834,0.230319596 C9.88637473,0.178773204 9.94372009,0.146556709 10.0068644,0.133670111 Z"
                fill-rule="nonzero"
              />
            </svg>
          </div>
          <div
            class="c2 modal-header"
          >
            <h5
              class="modal-title"
            >
              <span>
                Please confirm
              </span>
            </h5>
            <button
              aria-label="Close"
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            class="c3 modal-body"
          >
            <div>
              <img
                alt="icon"
                class="c4"
                src="IMAGE_MOCK"
              />
              <p
                class="c5 c6"
                color="lightOrange"
                font-size="md"
                font-weight="regular"
              >
                <span>
                  Are you sure you want to delete this?
                </span>
              </p>
            </div>
          </div>
          <div
            class="c7"
          >
            <button
              class="c8"
              color="delete"
              style="width: 100%;"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade"
    />
  </div>
</DocumentFragment>
`;
